src/login/LoginPage.jsx

```diff
@@ -46,28 +46,21 @@ import ResetPasswordSuccess from '../reset-password/ResetPasswordSuccess';
 const LoginPage = (props) => {
   const dispatch = useDispatch();
   const { formatMessage } = useIntl();
+  const backedUpFormData = useSelector(state => state.login.loginFormData);
+  const loginErrorCode = useSelector(state => state.login.loginErrorCode);
+  const loginErrorContext = useSelector(state => state.login.loginErrorContext);
+  const loginResult = useSelector(state => state.login.loginResult);
+  const shouldBackupState = useSelector(state => state.login.shouldBackupState);
+  const showResetPasswordSuccessBanner = useSelector(state => state.login.showResetPasswordSuccessBanner);
+  const submitState = useSelector(state => state.login.submitState);
```

Comment:
Have you noticed a decrease in rerendering the login component by granular selector state?

Reply:
There was an extra render of login component which was actually being caused by the re-rendering of its parent component (Logistration). 

Logistration was subscribed to register reducer thus a change in register redux state was causing the Logistration hence the Login component to re-render. 

By making the useSelectors granular in both components removed that extra rendered being caused by register store change.